armi/reactor/reactors.py

```diff
@@ -466,7 +463,8 @@ def removeAssembly(self, a1, discharge=True):
         self.remove(a1)
 
         if discharge and self._trackAssems:
-            self.sfp.add(a1)
+            if hasattr(self.parent, "sfp"):
+                self.parent.sfp.add(a1)
```

Comment:
Well, imagine we said:

```python
if self.parent is not None and  hasattr(self.parent, "sfp"):
```

It turns out `hasattr()` covers our bases so we don't need the "is not None" statement.

My major concerns are:

1.  This is Python, you can build a "Core" object in vaccuum with no "Reactor" object.
2. Maybe someone wants to build multiple cores for an analysis, that should be allowable with them having to build multiple reactors.
  a. People who complain about ARMI usually complain that everything is too complicated because they can't just do the one thing they want without: building a whole reactor or building an Operator/Plugin.  So I take it people would prefer they were freed from having to do so.

Reply:
Totally. I wasn't necessarily concerted with an `is not None` getting checked. I was thinking more about what happens in the case when there isn't a parent `Reactor` but a user still tries to add an assembly to `r.sfp`. In this current implementation, the user would never know that the assembly wasn't getting added as the code wouldn't tell them. The conditional would just evaluate to false and move on. So I was thinking maybe something akin to:

```python
if hasattr(self.parent, "sfp"):
    self.parent.sfp.add(a1)
else:
    runLog.warning(f"There is no parent Reactor object for {self}, so {a1} will not be added to the spent fuel pool")
```

I believe @keckler had a similar thought as well. https://github.com/terrapower/armi/pull/1336#discussion_r1253358584